rosetta/retriever/retriever_test.go

```diff
@@ -123,17 +115,20 @@ func TestRetriever_Current(t *testing.T) {
 			return mocks.GenericHeader, nil
 		}
 
-		ret, err := baselineRetriever(t)
-		require.NoError(t, err)
-
-		ret.index = index
+		ret := retriever.New(
+			mocks.GenericParams,
+			index,
+			mocks.BaselineValidator(t),
+			mocks.BaselineGenerator(t),
+			mocks.BaselineInvoker(t),
+			mocks.BaselineConverter(t),
+		)
 
 		blockID, blockTime, err := ret.Current()
 
-		if assert.NoError(t, err) {
-			assert.Equal(t, mocks.GenericHeader.ID().String(), blockID.Hash)
-			assert.Equal(t, mocks.GenericHeader.Timestamp, blockTime)
-		}
+		require.NoError(t, err)
+		assert.Equal(t, mocks.GenericHeader.ID().String(), blockID.Hash)
+		assert.Equal(t, mocks.GenericHeader.Timestamp, blockTime)
```

Comment:
Would be better at the top, no?

```go
header := mocks.GenericHeader
hash := header.ID().Hex()     // instead of `String()`
timestamp := header.Timestamp // maybe
```